apisix/plugins/example-plugin.lua

```diff
@@ -107,6 +116,49 @@ function _M.access(conf, ctx)
     return
 end
 
+
+function _M.before_proxy(conf, ctx)
+    core.log.warn("plugin before_proxy phase, conf: ", core.json.encode(conf))
+
+    if ctx.lua_proxy_upstream then
+        local status, body = lua_proxy_request(conf, ctx)
+        if status ~= 200 then
+            return status, body
+        end
+        core.log.warn("lua proxy upstream response: ", core.json.encode(body))
+        plugin_lua_body_filter(conf, ctx, body)
+    end
+end
+
+
+function _M.lua_body_filter(conf, ctx, body)
+    core.log.warn("plugin lua_body_filter phase, conf: ", core.json.encode(conf))
+    core.log.warn("plugin lua_body_filter phase, body: ", core.json.encode(body))
```

Comment:
you can assert those log in test cases to confirm the body come from upstream is passed to `lua_body_filter` as expect.

Reply:
ditto